sequencers/modeshape-sequencer-epub/src/main/java/org/modeshape/sequencer/epub/EpubMetadata.java

```diff
@@ -0,0 +1,317 @@
+            if (propertyName.equals("identifier")) {
+                getIdentifier().add(prop);
+            } else if (propertyName.equals("title")) {
+                getTitle().add(prop);
+            } else if (propertyName.equals("language")) {
+                getLanguage().add(prop);
+            } else if (propertyName.equals("contributor")) {
+                getContributor().add(prop);
+            } else if (propertyName.equals("creator")) {
+                getCreator().add(prop);
+            } else if (propertyName.equals("description")) {
+                getDescription().add(prop);
+            } else if (propertyName.equals("publisher")) {
+                getPublisher().add(prop);
+            } else if (propertyName.equals("rights")) {
+                getRights().add(prop);
+            } else if (propertyName.equals("date")) {
+                getDate().add(prop);
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Parse the container file to get the list of all rootfile packages.
+     */
+    private List<String> getRootfiles( ZipInputStream zipStream ) throws Exception {
+        List<String> rootfiles = new ArrayList<String>();
+        ZipEntry entry = null;
+        while ((entry = zipStream.getNextEntry()) != null) {
+            String entryName = entry.getName();
+            if (entryName.endsWith("META-INF/container.xml")) {
+                ByteArrayOutputStream content = getZipEntryContent(zipStream, entry);
+
+                DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+                DocumentBuilder builder = factory.newDocumentBuilder();
+                Document doc = builder.parse(new ByteArrayInputStream(content.toByteArray()));
+
+                XPathFactory xPathfactory = XPathFactory.newInstance();
+                XPath xpath = xPathfactory.newXPath();
+                XPathExpression expr = xpath.compile("/container/rootfiles/rootfile");
+                NodeList rootfileNodes = (NodeList) expr.evaluate(doc, XPathConstants.NODESET);
+
+                for (int i = 0; i < rootfileNodes.getLength(); i++) {
+                    Node node = rootfileNodes.item(i);
+                    rootfiles.add(node.getAttributes().getNamedItem("full-path").getNodeValue());
+                }
+                break;
+            }
+        }
+        return rootfiles;
+    }
+
+    /**
+     * Read the content of the ZipEntry without closing the stream.
+     */
+    private ByteArrayOutputStream getZipEntryContent(
+            ZipInputStream zipStream,
+            ZipEntry entry ) throws IOException {
+        try (ByteArrayOutputStream content =
+                new ByteArrayOutputStream()) {
+            byte[] bytes = new byte[(int) entry.getSize()];
+            int read;
+            while ((read = zipStream.read(bytes, 0, bytes.length)) != -1) {
+                content.write(bytes, 0, read);
+            }
+            content.flush();
```

Comment:
this is not required; the `ByteArrayOutputStream#flush` method is empty anyway. Writing to a `ByteArrayOutputStream ` is done immediately